pyproject.toml

```
@@ -0,0 +1,172 @@
+[build-system]
+requires = ["setuptools>=62.6", "versioneer[toml]==0.28"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "dask"
+description = "Parallel PyData with Task Scheduling"
+maintainers = [{name = "Matthew Rocklin", email = "mrocklin@gmail.com"}]
+license = {text = "BSD"}
+keywords = ["task-scheduling parallel numpy pandas pydata"]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Topic :: Scientific/Engineering",
+    "Topic :: System :: Distributed Computing",
+]
+readme = "README.rst"
+urls = {Homepage = "https://github.com/dask/dask/"}
+requires-python = ">=3.8"
+dependencies = [
+    # NOTE: These are tested in `continuous_integration/test_imports.sh` If
+    # you modify these, make sure to change the corresponding line there.
+    "click >= 7.0",
+    "cloudpickle >= 1.1.1",
+    "fsspec >= 0.6.0",
+    "packaging >= 20.0",
+    "partd >= 1.2.0",
+    "pyyaml >= 5.3.1",
+    "toolz >= 0.8.2",
+    "importlib_metadata >= 4.13.0",
+]
+dynamic = ["version"]
+
+[project.optional-dependencies]
+# NOTE: These are tested in `continuous_integration/test_imports.sh` If
+# you modify these, make sure to change the corresponding line there.
+array = ["numpy >= 1.21"]
+# keeping for backwards compatibility
+bag = []
+dataframe = [
+    "numpy >= 1.21",
+    "pandas >= 1.3",
+]
+distributed = ["distributed == 2023.3.1"]
+diagnostics = [
+    "bokeh >= 2.4.2, <3",
+    "jinja2 >= 2.10.3",
+]
+# keeping for backwards compatibility
+delayed = []
+complete = [
+    "dask[dataframe,distributed,diagnostics]",
```

Currently `array` is a subset of `dataframe` but it's possible that may not be the case in the future 

```suggestion
    "dask[array,dataframe,distributed,diagnostics]",
```